rust/theoros/src/storage/mod.rs

```diff
@@ -1,80 +1,79 @@
-pub mod event;
+pub mod checkpoints;
 pub mod feed_id;
+pub mod updates;
 pub mod validator;
 
-pub use event::*;
+pub use checkpoints::*;
 pub use feed_id::*;
-use tokio::sync::broadcast::Sender;
+pub use updates::*;
 pub use validator::*;
 
 use starknet::core::types::Felt;
+use tokio::sync::broadcast::Sender;
 
 use crate::{
     rpc::starknet::{HyperlaneCalls, PragmaFeedsRegistryCalls, StarknetRpc},
-    types::hyperlane::CheckpointMatchEvent,
+    types::hyperlane::NewUpdatesAvailableEvent,
 };
 
-/// Theoros storage that contains:
-///   * a set of all available feed ids,
-///   * a mapping of all the validators and their fetchers.
-///   * a mapping of all the validators and their latest fetched checkpoints.
-///   * an event cache,
-///   * an events storage containing the most recents [DispatchEvent] events indexed.
-///   * a channel to dispatch updates to the clients.
 pub struct TheorosStorage {
     feed_ids: FeedIdsStorage,
-    validators: ValidatorsLocationStorage,
-    checkpoints: ValidatorsCheckpointsStorage,
-    cached_events: EventCache,
-    dispatch_events: EventStorage,
-    pub feeds_channel: Sender<CheckpointMatchEvent>,
+    validators_fetchers: ValidatorsFetchersStorage,
+    signed_checkpoints: SignedCheckpointsStorage,
+    unsigned_checkpoints: UnsignedCheckpointsStorage,
+    latest_update_per_feed: LatestUpdatePerFeedStorage,
+    // websocket notifications
+    feeds_updated_tx: Sender<NewUpdatesAvailableEvent>,
 }
 
 impl TheorosStorage {
     pub async fn from_rpc_state(
```

Comment:
a better way to implement that would to have a `FromRpc` trait and implement it on this struct and underlying struct like `FeedIdsStorage`. Although it's probably not worth the complexity here it's worth noting.

Reply:
Agreed, definitely something to add if we have more structs working this way